File: leetcode38.cpp
class Solution {
public:
    char IntToChar(int n)
    {
        char res;
        res = n + 48;
        return res;
    }
    string countAndSay(int n) {
        if(n < 0) return "";
        if(n == 1) return "1";
        string result = countAndSay(n-1);
        int count = 1;
        string s = "";
        char tmp;
        for (int i = 0; i < result.size() - 1; ++i)
        {
        	if(result[i] == result[i+1])
        		count++;
        	else
        	{
        		tmp = IntToChar(count); 
        		s.push_back(tmp);
				s.push_back(result[i]);
        		count = 1;
        	}
        }
        return s;
    }
};

File: leetcode38.java
public class Solution {
	public String helper(String str){
		char c = str.charAt(0);
		StringBuffer res = new StringBuffer();
		int count = 1;
		for(int i = 1; i < str.length(); i++){
			if(str.charAt(i) == c){
				count++;
			}
			else{
				res.append(count);
				res.append(c);
				c = str.charAt(i);
				count = 1;
			}			
		}
		res.append(count);
		res.append(c);
		return res.toString();
	}

    public String countAndSay(int n) {
        String s = "1";
        for(int i = 1; i < n; ++i){
        	s = helper(s);
        }
        return s;
    }
}
